maximoortelli/OOP_SCHOOL_LIBRARY_PART_3
# Add all files

### Description:
-----------------------------------------------------------------------------------------------------------------------------------------

_This PR aims to merge the tetsing-with-rspec feature into the develop branch. In this feature, unit tests have been added to each class in the project, in accordance with the specified requirements.
(Please read the bottom lines with Marco Lizza permission)_

-----------------------------------------------------------------------------------------------------------------------------------------
### General Requirements:
-----------------------------------------------------------------------------------------------------------------------------------------

- [x] No linter errors in the codebase.
- [x] Follow the correct Gitflow.
- [x] Document all changes and logic in a professional manner.

-----------------------------------------------------------------------------------------------------------------------------------------
### Ruby Requirements:
-----------------------------------------------------------------------------------------------------------------------------------------

- [x] The codebase follows the list of best practices for Ruby as specified.

-----------------------------------------------------------------------------------------------------------------------------------------
### Project Specific Requirements:
-----------------------------------------------------------------------------------------------------------------------------------------

- Add unit tests for the following classes:

Book

Person

Student

Teacher

Classroom

Rental

- All classes related to the implementation of the Decorator design pattern

-----------------------------------------------------------------------------------------------------------------------------------------
### Changes:
-----------------------------------------------------------------------------------------------------------------------------------------

- File: book_spec.rb
- Add unit tests for all methods in the Book class.
- File: person_spec.rb
- Add unit tests for all methods in the Person class.
- File: student_spec.rb
- Add unit tests for all methods in the Student class.
- File: teacher_spec.rb
- Add unit tests for all methods in the Teacher class.
- File: classroom_spec.rb
- Add unit tests for all methods in the Classroom class.
- File: rental_spec.rb
- Add unit tests for all methods in the Rental class.
- File: decorators_spec.rb
- Add unit tests for all methods in all the classes related to the decorator pattern.
- 
-----------------------------------------------------------------------------------------------------------------------------------------
### How to Test:
-----------------------------------------------------------------------------------------------------------------------------------------

_To test the changes, you can add the Rspec gem and run the bundle install command on your terminal after running the following command:_

_rspec
This should run all the unit tests and they should all pass.

I have been given permission by the Student Success team to work solo this week. If you have any questions please reach out to Marco Lizza via the Code Reviewers Slack workspace._
